pkg/bootstrapper/bootstrapper.go

```diff
@@ -558,7 +564,21 @@ func (wmcb *winNodeBootstrapper) generateInitialKubeletArgs(args map[string]stri
 	if wmcb.nodeIP != "" {
 		kubeletArgs = append(kubeletArgs, "--node-ip="+wmcb.nodeIP)
 	}
-	return kubeletArgs
+
+	var hostname string
+	var err error
+	switch wmcb.platformType {
+	case "AWS":
+		hostname, err = getAWSMetadataHostname()
+		if err != nil {
+			return nil, err
+		}
+	}
```

Comment:
This switch could grow over time, so as not to distract from the purpose of this function, why not make this a helper?

```
func getKubeletHostname(platform configv1.PlatformType) (string, error) {
  switch platform {
  case configv1.AWSPlatformType:
    return getAWSMetadataHostname()
  default:
    return "", nil
  }
}
```

Reply:
Agreed, I'd also like to see this moved to a separate package

Reply:
moved to a new package "cloud"